dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedIntervalIterableTest.java

```diff
@@ -86,23 +85,25 @@ private void assertIterations(
   public void testHours() {
     Duration durationOfDay = Duration.ofDays(1);
     IntervalExpander dayExpander = IntervalExpander.createBasedOnDuration(durationOfDay);
-
-    Duration durationOfHour = Duration.ofDays(1);
-    IntervalExpander hourExpander = IntervalExpander.createBasedOnDuration(durationOfDay);
-
-    testIterable(
-        169,
-        ZonedIntervalIterableGenerator.forTimeUnitsUntilNow(24 * 7, durationOfDay, dayExpander));
-    testIterable(
-        8, ZonedIntervalIterableGenerator.forTimeUnitsUntilNow(7, durationOfDay, dayExpander));
-
-    testIterable(
-        1,
+    Duration durationOfHour = Duration.ofHours(1);
+    IntervalExpander hourExpander = IntervalExpander.createBasedOnDuration(durationOfHour);
+    ZonedIntervalIterable weekInHours =
+        ZonedIntervalIterableGenerator.forTimeUnitsUntilNow(24 * 7, durationOfHour, hourExpander);
+    ZonedIntervalIterable weekInDays =
+        ZonedIntervalIterableGenerator.forTimeUnitsUntilNow(7, durationOfDay, dayExpander);
+    ZonedIntervalIterable lastHourBefore2020 =
         ZonedIntervalIterableGenerator.forTimeUnitsUntil(
             ZonedDateTime.of(2020, 1, 1, 0, 0, 0, 0, ZoneId.of("UTC")),
             1,
             durationOfHour,
-            hourExpander));
+            hourExpander);
+    ZonedIntervalIterable dayInHours =
+        ZonedIntervalIterableGenerator.forTimeUnitsUntilNow(24, durationOfHour, hourExpander);
+
+    // Act + Assert
+    testIterable(8, weekInDays);
```

Comment:
I'm a bit confused. There is 7 days in a week, isn't ?

Reply:
Yes, that didn't change ;)
The logic of `ZonedIntervalIterable` which is tested here is a bit odd. The idea is that we have time separated into segments (days or hours) and we receive an arbitrary period of time.

The goal is to select the minimum set of segments that covers our time period. Let's say that the time period is until now (13:30 on Thursday), from exactly a week ago and we would like to use segment size = 1 day.

This is 7 days, but creates 8 segments - we have to select all of this Thursday, all of last Thursday and the 6 one-day segments in between. OTOH it will be 7 segments if run exactly at midnight.

Readability of the test is probably not much improved by how this method call looks a bit like an assertion, with `expected=8, actual=weekInDays`. I'd write these tests differently, but I wanted to limit this PR to bug fixes, without any rewriting.